### ISSUE TITLE
Some mount options ignored in 1.8.6

### ISSUE THREAD

**Issue by taisph** [mitchellh/vagrant#7859]
### Vagrant version

Vagrant 1.8.6
### Host operating system

Distributor ID: Ubuntu
Description:    Ubuntu 16.04.1 LTS
Release:    16.04
Codename:   xenial
### Guest operating system

Distributor ID: CentOS
Description:    CentOS Linux release 7.2.1511 (Core)
Release:    7.2.1511
Codename:   Core
### Vagrantfile

``` ruby
  config.vm.synced_folder ".", "/vagrant", owner: "vagrant", group: "vagrant", :mount_options => ["uid=1001", "gid=1001", "dmode=775", "umask=002"]
```
### Expected behavior

Share mounted with uid=1001 and gid=1001 as it did in 1.8.5 and earlier.
### Actual behavior

Share mounted with uid=1000 and gid=1000 (vagrant).

Owner/group options doesn't accept uid/gids and the user who needs to own the files is created during provisioning, so it is not possible to use the owner/group options.

**Comment by adejongh** [comment]
If I understand the codebase correctly, this commit may be the culprit here:
https://github.com/mitchellh/vagrant/commit/dd6ad2fbf3bc935dbb127bf5552dcc5990b8e3e6

Looking at the history of mount_virtualbox_shared_folder.rb that is also the only commit since the previous release.

**Comment by dol** [comment]
The bug was introduced in #7720

**Comment by dol** [comment]
I would recommend to skipping user/group name uid/gid translation if the mount option uid or gid is passed.

**Comment by chrisroberts** [comment]
Fixed via #8122. Thanks for reporting the issue!
